cc/platform/utils_osx.cc

```diff
@@ -32,6 +29,10 @@ int GetNumLogicalCpus() {
   return nproc;
 }
 
+bool FdSupportsAnsiColors(int fd) { return isatty(fd); }
+
+ProcessId GetProcessId() { return getpid(); }
+
```

Comment:
for two lines below (GetHostname) should you be initializing hostname = {}? if the if condition isn't triggered isn't it possible that hostname doesn't contain a null character and std::string() reads past end of hostname?

Reply:
I've cleaned that code up. I think it was ok because if the `gethostname` call failed, the code would copy the string literal `"hostname"` into the buffer before converting it to a string.

You made me reread the `gethostname` docs for windows and posix because I was worried about corner cases, which of course there are. Fortunately, it seems like using a 256B buffer avoids them :)